crates/mun_codegen_macros/src/lib.rs

```diff
@@ -173,37 +247,82 @@ pub fn as_value_derive(input: TokenStream) -> TokenStream {
                         let type_context = context.type_context;
                         let field_padding = #field_padding_values;
 
-                        let struct_type = Self::get_ir_type(context.type_context);
+                        if <#ident>::has_const_value() {
```

Comment:
What does this mean in this context? It would be good to add a comment here on the two paths.

Reply:
I forgot this, I think. If everything else is fine, I'll include this in the final rebase submission

Reply:
Resolved.